apps/language_server/lib/language_server/providers/document_symbols.ex

```diff
@@ -169,14 +169,18 @@ defmodule ElixirLS.LanguageServer.Providers.DocumentSymbols do
 
   # Types
   defp extract_symbol(_current_module, {:@, location, [{type_kind, _, type_expression}]})
-       when type_kind in [:type, :typep, :opaque, :callback, :macrocallback] do
+       when type_kind in [:type, :typep, :opaque, :callback, :macrocallback] and
```

Comment:
this is getting pretty complicated, can we do the nill check in a separate function head before this one?

```elixir
defp extract_symbol(_current_module, {:@, location, [{type_kind, _, nil}]}) do
  ...
end
```

Reply:
Then we would have attribute handling in 2 clauses - one with `nil` and one lower matching `_`. Rejecting `nil` here make it fall back to the lower `_` clause handling all non typespec attributes